llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -1460,7 +1460,15 @@ bool SIFoldOperands::tryFoldFoldableCopy(
     return false;
   }
 
-  MachineOperand &OpToFold = MI.getOperand(1);
+  MachineOperand *OpToFoldPtr;
+  if (MI.getOpcode() == AMDGPU::V_MOV_B16_t16_e64) {
```

Comment:
I don't expect to need special casing here. Should this be filtered in isFoldableCopy? I thought we had filtering for regular v_mov_b32_e64 already somewhere here

Reply:
Hi Matt. v_mov_b32_e64 only has 1 source operand so we don't the filtering.  
V_MOV_B16_t16_e64 has three input operands and  I think it is only instruction in the FoldableCopy list that has more than 1 operand:
`inoperandlist: src_modifers, src0, op_sel`
so it need some special handling here.

I think we could move the filtering to isFoldableCopy though, but we need still need the special handling to select correct operand. What do you think?